axum-extra/src/routing/spa.rs

```diff
@@ -201,15 +202,173 @@ where
     }
 }
 
+pub trait SpaRouterExt {
+    fn spa_assets(self, route: &str, path: impl AsRef<Path>) -> Self;
+}
+
+impl SpaRouterExt for Router {
+    fn spa_assets(self, route: &str, path: impl AsRef<Path>) -> Self {
+        let mut serve_assets = get_service(ServeDir::new(&path))
+            .handle_error(|_| async { StatusCode::INTERNAL_SERVER_ERROR });
+        let mut serve_index = spa_index(&path);
+
+        let handler = move |req: Request<Body>| async move {
+            let req_clone = {
+                let mut clone = Request::new(Body::default());
+                *clone.method_mut() = req.method().clone();
+                *clone.uri_mut() = req.uri().clone();
+                *clone.headers_mut() = req.headers().clone();
+                clone
+            };
+
+            let res = serve_assets.call(req).await.into_response();
+
+            if res.status() == StatusCode::NOT_FOUND {
+                serve_index.call(req_clone).await.into_response()
+            } else {
+                res
+            }
+        };
+
+        self.nest(route, handler.into_service())
```

Comment:
This wont work if `route` is `/`. See https://github.com/tokio-rs/axum/issues/933#issuecomment-1104819601

Reply:
Ah right, I forgot about this. Will see how I can rewrite this with a `.merge`. Does that mean, in the original `SpaRouter` the `.fallback` call is not propagated to the _outer_ router?
So the fallback only applies to the inner SPA router rather than registering a global fallback in the outer one?

Reply:
You'll end up with the same design I had 😅 at least I don't see how it's different